feature-libs/pickup-in-store/components/container/set-preferred-store/set-preferred-store.component.html

```diff
@@ -1,30 +1,64 @@
-<div
-  [attr.data-preferred-store]="pointOfServiceName.name"
-  [attr.data-store-is-selected]="
-    pointOfServiceName.name === (storeSelected$ | async)?.name
-  "
-  class="setpreferredstore-container"
-  (click)="setAsPreferred()"
->
+<ng-container *cxFeature="'a11yRepeatingButtonsUniqueLabels'">
   <div
-    [ngClass]="{
-      'icon-selected':
-        pointOfServiceName.name === (storeSelected$ | async)?.name,
-      'icon-not-selected':
-        pointOfServiceName.name !== (storeSelected$ | async)?.name,
-    }"
+    *ngIf="storeSelected$ | async as storeSelected"
+    [attr.data-preferred-store]="pointOfServiceName.name"
+    [attr.data-store-is-selected]="
+      pointOfServiceName.name === storeSelected.name
+    "
+    class="setpreferredstore-container"
+    (click)="setAsPreferred()"
   >
-    <cx-icon aria-hidden="true" [type]="ICON_TYPE.HEART"></cx-icon>
+    <div
+      [ngClass]="{
+        'icon-selected': pointOfServiceName.name === storeSelected.name,
+        'icon-not-selected': pointOfServiceName.name !== storeSelected.name,
+      }"
+    >
+      <cx-icon aria-hidden="true" [type]="ICON_TYPE.HEART"></cx-icon>
+    </div>
+    <button
+      data-text="setPreferredStore.myStore"
+      [attr.data-preferred-store]="pointOfServiceName.name"
+      class="set-preferred-heading"
+      [attr.aria-label]="
+        (getSetStoreButtonLabel(storeSelected.name) | cxTranslate) +
+        ', ' +
+        pointOfServiceName.displayName
+      "
+    >
+      {{ getSetStoreButtonLabel(storeSelected.name) | cxTranslate }}
+    </button>
   </div>
-  <button
-    data-text="setPreferredStore.myStore"
+</ng-container>
+<ng-container *cxFeature="'a11yRepeatingButtonsUniqueLabels'">
+  <div
     [attr.data-preferred-store]="pointOfServiceName.name"
-    class="set-preferred-heading"
-  >
-    {{
+    [attr.data-store-is-selected]="
       pointOfServiceName.name === (storeSelected$ | async)?.name
-        ? ('setPreferredStore.myStore' | cxTranslate)
-        : ('setPreferredStore.makeThisMyStore' | cxTranslate)
-    }}
-  </button>
-</div>
+    "
+    class="setpreferredstore-container"
+    (click)="setAsPreferred()"
+  >
+    <div
+      [ngClass]="{
+        'icon-selected':
+          pointOfServiceName.name === (storeSelected$ | async)?.name,
+        'icon-not-selected':
+          pointOfServiceName.name !== (storeSelected$ | async)?.name,
+      }"
```

Comment:
Leftover `(storeSelected$ | async)?.name`

```suggestion
     [ngClass]="pointOfServiceName.name === storeSelected.name ? 'icon-selected' : 'icon-not-selected'"
```

Reply:
We should leave the old code as is to avoid breaking changes.